/*
 * This file was automatically generated by EvoSuite
 * Thu Sep 21 09:11:34 GMT 2023
 */

package org.example;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.example.Trivialissimo;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true) 
public class Trivialissimo_ESTest extends Trivialissimo_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      int int0 = Trivialissimo.soma(0, 1);
      assertEquals(1, int0);
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      int int0 = Trivialissimo.soma((-3794), 1564);
      assertEquals((-2230), int0);
  }

  @Test(timeout = 4000)
  public void test2()  throws Throwable  {
      int int0 = Trivialissimo.divisao((-1), 1762);
      assertEquals(0, int0);
  }

  @Test(timeout = 4000)
  public void test3()  throws Throwable  {
      int int0 = Trivialissimo.divisao((-1196), (-622));
      assertEquals(1, int0);
  }

  @Test(timeout = 4000)
  public void test4()  throws Throwable  {
      int int0 = Trivialissimo.divisao((-1860), 1404);
      assertEquals((-1), int0);
  }

  @Test(timeout = 4000)
  public void test5()  throws Throwable  {
      // Undeclared exception!
      try { 
        Trivialissimo.divisao(0, 0);
        fail("Expecting exception: ArithmeticException");
      
      } catch(ArithmeticException e) {
         //
         // / by zero
         //
         verifyException("org.example.Trivialissimo", e);
      }
  }

  @Test(timeout = 4000)
  public void test6()  throws Throwable  {
      Trivialissimo trivialissimo0 = new Trivialissimo();
  }

  @Test(timeout = 4000)
  public void test7()  throws Throwable  {
      int int0 = Trivialissimo.soma(0, 0);
      assertEquals(0, int0);
  }
}
